source/grid/grid_generator.cc

```diff
@@ -8403,6 +8403,209 @@ namespace GridGenerator
 
 
 
+  template <int dim, int spacedim>
+  void
+  alfeld_split_of_simplex_mesh(const Triangulation<dim, spacedim> &in_tria,
+                               Triangulation<dim, spacedim>       &out_tria)
+  {
+    Assert(dim == 2, ExcNotImplemented());
+
+    Triangulation<dim, spacedim> temp_tria;
+    if (in_tria.n_global_levels() > 1)
+      {
+        AssertThrow(!in_tria.has_hanging_nodes(), ExcNotImplemented());
+        GridGenerator::flatten_triangulation(in_tria, temp_tria);
+      }
+    const Triangulation<dim, spacedim> &ref_tria =
+      in_tria.n_global_levels() > 1 ? temp_tria : in_tria;
+
+    // Three triangles connecting to barycenter with vertex index 3:
+    static const ndarray<unsigned int, 3, 3> table_2D_cell = {
+      {{{0, 1, 3}}, {{1, 2, 3}}, {{2, 0, 3}}}};
+
+    /* Boundary-faces 2d:
+     * After converting, each of the 4 quadrilateral faces is defined by faces
+     * of 2 different triangles, i.e., lines. Note that lines are defined by 2
+     * vertices.
+     */
```

Comment:
Use `//`-style comments like everywhere else.

Reply:
Separately, I don't understand the comment. Is this a copy-paste mistake from elsewhere?